src/Plugin/LanguageNegotiation/LanguageNegotiationAdmin.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Drupal\helfi_platform_config\Plugin\LanguageNegotiation;
+
+use Drupal\language\LanguageNegotiationMethodBase;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class for identifying language from the user preferences.
+ *
+ * @LanguageNegotiation(
+ *   id = \Drupal\helfi_platform_config\Plugin\LanguageNegotiation\LanguageNegotiationAdmin::METHOD_ID,
+ *   weight = -4,
+ *   name = @Translation("Admin"),
+ *   description = @Translation("Follow the user's administration language preference.")
+ * )
+ */
+class LanguageNegotiationAdmin extends LanguageNegotiationMethodBase {
+
+  /**
+   * The language negotiation method id.
+   */
+  const METHOD_ID = 'language-admin-preference';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLangcode(Request $request = NULL) {
+    $langcode = NULL;
+
+    if ($this->languageManager && $this->currentUser->isAuthenticated()) {
```

Comment:
We should check whether the user has access to administrator pages here since the setting is only available for users with `access administration pages` or `view the administration theme` permissions:
![image](https://user-images.githubusercontent.com/771113/229024662-86234564-5ff1-4cce-914d-b0fbca3adfeb.png)

Reply:
Actually, we probably should add a form alter hook to expose that setting when `language-admin-preference` negotiation method is enabled.